corehq/apps/accounting/bootstrap/utils.py

```diff
@@ -208,6 +163,84 @@ def _ensure_feature_rates(feature_rates, features, edition, verbose, apps):
     return db_feature_rates
 
 
+def _ensure_software_plan(plan_key, product, product_rate, verbose, apps):
+    SoftwarePlan = apps.get_model('accounting', 'SoftwarePlan')
+    plan_name = _software_plan_name(plan_key, product, product_rate)
+    try:
+        software_plan = SoftwarePlan.objects.get(name=plan_name)
+        if verbose:
+            log_accounting_info(
+                "Plan '%s' already exists. Using existing plan to add version." % software_plan.name
+            )
+    except SoftwarePlan.DoesNotExist:
+        plan_opts = {
+            'name': plan_name,
+            'edition': plan_key.edition,
+            'visibility': (SoftwarePlanVisibility.ANNUAL
+                           if plan_key.is_annual_plan else SoftwarePlanVisibility.PUBLIC),
+        }
+        if plan_key.is_annual_plan is not None:
+            plan_opts['is_annual_plan'] = plan_key.is_annual_plan
+        software_plan = SoftwarePlan.objects.create(**plan_opts)
+        if verbose:
+            log_accounting_info("Creating Software Plan: %s" % software_plan.name)
+    return software_plan
+
+
+def _software_plan_name(plan_key, product, product_rate):
+    plan_name = (
+        (('%s Trial' % product_rate.name) if plan_key.is_trial else ('%s Edition' % product_rate.name))
```

Comment:
I know this is copied from very old code, but can we use f-strings to make this more readable?
```suggestion
        (f'{product_rate.name} Trial' if plan_key.is_trial else f'{product_rate.name} Edition')
```

Reply:
Could this need to be translated, or is this always used for internal representations like `log_accounting_info`? If it may be translated then f-strings cannot be used, although `string.format()` is still better than the old `%` syntax in most cases.

Reply:
As far as I can see it's only ever used internally here, since we're using it to look up and/or generate model instances. Since it does get shown to a user in limited cases like renewing their plan, it could be that it's translated later on (I don't think so, but it could be), but I would think that would be based on the string saved on the model instance, not this string construction.